<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="es165_ros2_control" params="name">
    <!-- ROS 2 control block -->
    <ros2_control name="arm_controller" type="system">

      <!-- Isaac Sim hardware interface -->
      <hardware>
      <!-- This plugin name must match the Isaac ROS control plugin -->
        <!-- <plugin>topic_based_ros2_control/TopicBasedSystem</plugin>
        <param name="joint_commands_topic">isaac_joint_commands</param>
        <param name="joint_states_topic">isaac_joint_states</param> -->
        <plugin>mock_components/GenericSystem</plugin>
        <!-- <param name="fake_sensor_commands">true</param> -->
      </hardware>

      <!-- Joint interfaces -->
      <joint name="joint_1_s">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <state_interface name="effort"/> -->
      </joint>

      <joint name="joint_2_l">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <state_interface name="effort"/> -->
      </joint>

      <joint name="joint_3_u">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <state_interface name="effort"/> -->
      </joint>

      <joint name="joint_4_r">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <state_interface name="effort"/> -->
      </joint>

      <joint name="joint_5_b">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <state_interface name="effort"/> -->
      </joint>

      <joint name="joint_6_t">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <!-- <state_interface name="effort"/> -->
      </joint>

    </ros2_control>
  </xacro:macro>
</robot>
